tests/std/tests/P0645R10_text_formatting_formatting/test.cpp

```diff
@@ -8,252 +8,359 @@
 #include <string>
 #include <string_view>
 
+// copied from the string_view tests
+template <typename CharT>
+struct choose_literal; // not defined
+
+template <>
+struct choose_literal<char> {
+    static constexpr const char* choose(const char* s, const wchar_t*) {
+        return s;
+    }
+};
+
+template <>
+struct choose_literal<wchar_t> {
+    static constexpr const wchar_t* choose(const char*, const wchar_t* s) {
+        return s;
+    }
+};
+
+#define TYPED_LITERAL(CharT, Literal) (choose_literal<CharT>::choose(Literal, L##Literal))
+
+
 using namespace std;
 // TODO: fill in tests
 template back_insert_iterator<string> std::vformat_to(
     back_insert_iterator<string>, const locale&, string_view, format_args_t<back_insert_iterator<string>, char>);
 
+template <class charT, class... Args>
+auto make_testing_format_args(Args&&... vals) {
+    if constexpr (is_same_v<charT, wchar_t>) {
+        return make_wformat_args(forward<Args>(vals)...);
+    } else {
+        return make_format_args(forward<Args>(vals)...);
+    }
+}
 
-int main() {
-    string output_string = "";
 
-    vformat_to(back_insert_iterator(output_string), locale::classic(), "f", make_format_args());
-    assert(output_string == "f");
+// tests for format with no format args or replacement fields
+template <class charT>
+void test_simple_formatting() {
+    basic_string<charT> output_string;
+
+    vformat_to(back_insert_iterator(output_string), locale::classic(), TYPED_LITERAL(charT, "f"),

```

Comment:
Pre-existing: We try to use braces to construct temporaries like `back_insert_iterator` as this avoids looking like function calls. Occurs repeatedly.